tracker/graceful.go

```diff
@@ -0,0 +1,75 @@
+package tracker
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/Wessie/fdstore"
+)
+
+type wireRecorder struct {
+	Listeners map[radio.ListenerClientID]*Listener
+}
+
+func (s *Server) storeSelf(ctx context.Context, fds *fdstore.Store) error {
+	err := fds.AddListener(s.httpLn, HttpLn, nil)
+	if err != nil {
+		return err
+	}
+	err = fds.AddListener(s.grpcLn, GrpcLn, nil)
+	if err != nil {
+		return err
+	}
+	return s.recorder.storeSelf(ctx, fds)
+}
+
+func (r *Recorder) storeSelf(ctx context.Context, fds *fdstore.Store) error {
```

Comment:
Either just make these generic `UnmarshalJSON`/`MarshalJSON` functions on the recorder and call them from the server store function or use the protobuf types in `rpc`, either works